### DISCUSSION TITLE
pydantic serialization_alias not work as expected in swagger while model is set in parameter "responses"

### DISCUSSION THREAD

**Discussion by chenxin6** [fastapi/fastapi#11717] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
router = APIRouter(tags=["bot"], prefix="/v2")

class GetBotResponseNew(BaseModel):
    user_id: str = Field(..., serialization_alias="userId")
    bot_id: str = Field(..., serialization_alias="botId")
    bot_name: str = Field(..., serialization_alias="botName")

@router.get("/bot/{botId}",
            responses={
                200: {"model": GetBotResponseNew},
                403: {"model": GetBotResponseNewFor403},
                404: {"model": GetBotResponseNewFor404}
            })
def get_bot(bot_id: str = Path(..., description="the id of the bot", alias="botId"),
            user: UserResponse = Depends(get_current_user)) -> GetBotResponseNew:
    return get_bot_internal(bot_id=bot_id, user=user)
```

### Description

![image](https://github.com/tiangolo/fastapi/assets/40751203/4eeb9d19-f398-4899-b63a-5097377d8ae4)

The parameters like `user_id`, `bot_id`, `bot_name`  in swagger are not in camel case.

### Operating System

Linux

### FastAPI Version

0.110.0

### Pydantic Version

2.6.4

### Python Version

Python 3.11.7

**Comment by chenxin6** [comment]
I noticed that the model name in swagger is `GetBotResponseNew-Input` instead of `GetBotResponseNew`. Below is what I found in official docs but I still don't know how to fix it.

```
    json_schema_mode_override: Literal['validation', 'serialization', None]

    If not `None`, the specified mode will be used to generate the JSON schema regardless of what `mode` was passed to
    the function call. Defaults to `None`.

    This provides a way to force the JSON schema generation to reflect a specific mode, e.g., to always use the
    validation schema.

    It can be useful when using frameworks (such as FastAPI) that may generate different schemas for validation
    and serialization that must both be referenced from the same schema; when this happens, we automatically append
    `-Input` to the definition reference for the validation schema and `-Output` to the definition reference for the
    serialization schema. By specifying a `json_schema_mode_override` though, this prevents the conflict between
    the validation and serialization schemas (since both will use the specified schema), and so prevents the suffixes
    from being added to the definition references.
```

**Comment by nonstop** [comment]
If you call **model_dump** from inside **get_bot_internal**
replace it with **model_dump(by_alias=True)**.
This fixes the problem.

**Comment by YuriiMotov** [comment]
This works well in current version of FastAPI.
